lambda/translattribute.ml

```diff
@@ -17,40 +17,38 @@ open Typedtree
 open Lambda
 open Location
 
-let is_inline_attribute = function
-  | {txt=("inline"|"ocaml.inline")} -> true
-  | _ -> false
 
-let is_inlined_attribute = function
-  | {txt=("inlined"|"ocaml.inlined")} -> true
-  | {txt=("unrolled"|"ocaml.unrolled")} when Config.flambda || Config.flambda2 -> true
-  | _ -> false
+let is_inline_attribute =
+  [ ["inline"; "ocaml.inline"],true ]
 
-let is_specialise_attribute = function
-  | {txt=("specialise"|"ocaml.specialise")} when Config.flambda -> true
-  | _ -> false
+let is_inlined_attribute =
+  [ ["inlined"; "ocaml.inlined"], true
+  ; ["unrolled"; "ocaml.unrolled"], (Config.flambda || Config.flambda2)
+  ]
 
-let is_specialised_attribute = function
-  | {txt=("specialised"|"ocaml.specialised")} when Config.flambda -> true
-  | _ -> false
+let is_specialise_attribute =
+  [ ["specialise"; "ocaml.specialise"], Config.flambda ]
 
-let is_local_attribute = function
-  | {txt=("local"|"ocaml.local")} -> true
-  | _ -> false
+let is_specialised_attribute =
+  [ ["specialised"; "ocaml.specialised"], Config.flambda ]
+
+let is_local_attribute =
+  [ ["local"; "ocaml.local"], true ]
+
+let is_tailcall_attribute =
+  [ ["tailcall"; "ocaml.tailcall"], true ]
 
 let find_attribute p attributes =
-  let inline_attribute, other_attributes =
-    List.partition (fun a -> p a.Parsetree.attr_name) attributes
-  in
+  let inline_attribute = Builtin_attributes.filter_attributes p attributes in
   let attr =
     match inline_attribute with
     | [] -> None
     | [attr] -> Some attr
-    | _ :: {Parsetree.attr_name = {txt;loc}; _} :: _ ->
+    | attr :: {Parsetree.attr_name = {txt;loc}; _} :: _ ->
       Location.prerr_warning loc (Warnings.Duplicated_attribute txt);
-      None
+      Some attr
```

Comment:
Switching from `None` to `Some attr` looks like a bugfix - did this really ignore attributes that were listed twice, or am I reading the old code wrong?

Reply:
I think the new behaviour is much better. Arbitrarily picking between two attributes isn't great, but in the old behaviour `[@inline][@inline]` means "don't inline", which is nonsense.

Also, the choice isn't completely arbitrary - I think parse order is preserved, so the attribute that wins is presumably whichever's first in the source code, which seems reasonable. (Or maybe it's last-wins, I haven't checked whether this list is built in reverse order. Last-wins is also fine by me).